UI/Editor/ArrayInspector.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using UnityEditor;
+using UnityEngine;
+
+namespace Dubit.DUCK.Editor.Inspectors
+{
+	public abstract class ArrayInspector : UnityEditor.Editor
+	{
+		protected abstract string arrayPropertyName
+		{
+			get;
+		}
+
+		protected virtual int arrayFoldoutBitmask
+		{
+			get
+			{
+				return -1;
+			}
+
+			set
+			{
+				// Default: do nothing
+			}
+		}
+
+		protected abstract Action<SerializedProperty> drawArrayElement
+		{
+			get;
+		}
+
+		protected SerializedProperty serializedArray;
+
+		protected void OnEnable()
+		{
+			serializedArray = serializedObject.FindProperty(arrayPropertyName);
+		}
+
+		public override void OnInspectorGUI()
+		{
+			serializedObject.Update();
+
+			if (serializedArray != null)
+			{
+				arrayFoldoutBitmask = DrawArray(serializedArray, drawArrayElement, arrayFoldoutBitmask);
+			}
+
+			serializedObject.ApplyModifiedProperties();
+		}
+
+		public static int DrawArray(SerializedProperty arrayProperty, Action<SerializedProperty> drawArrayElement, int selectionBitmask = -1)
+		{
+			if (arrayProperty == null || arrayProperty.arraySize < 0)
```

Comment:
Should be either same line or with a block

Reply:
Done